### PR TITLE
Fix some errors that occur when importing o.e.j.c incubator

### PR DESCRIPTION
**Pull request by datho7561**
- Do not reuse the context when compiling to multiple output directories

### CODE REVIEW

**Comment by datho7561** [comment]
~~All the regressions are caused by 'possible this escape' errors, which are now reported~~

update: there should be no test regressions caused by this PR

**Comment by mickaelistria** [comment]
Can you please split it into 2 distinct commits for each part of the change? I think the context bit is more risky than the problems mapping bit there are chances we'll have to revert it. Smaller commits allows less loss during reverts.

**Comment by datho7561** [comment]
I created a separate PR for the problem id changes.

**Comment by datho7561** [comment]
test this please

File: org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacBindingResolver.java [hunk 1]
```diff
@@ -939,7 +939,11 @@ IMethodBinding resolveConstructor(SuperConstructorInvocation expression) {
 			javacElement = javacMethodInvocation.getMethodSelect();
 		}
 		if (javacElement instanceof JCIdent ident && ident.sym instanceof MethodSymbol methodSymbol) {
-			return this.bindings.getMethodBinding(ident.type != null ? ident.type.asMethodType() : methodSymbol.asType().asMethodType(), methodSymbol, null, false);
+			if (ident.type != null && (ident.type instanceof MethodType || ident.type instanceof ForAll)) {
+				return this.bindings.getMethodBinding(ident.type.asMethodType(), methodSymbol, null, false);
+			} else if (methodSymbol.asType() instanceof MethodType || methodSymbol.asType() instanceof ForAll) {
+				return this.bindings.getMethodBinding(methodSymbol.asType().asMethodType(), methodSymbol, null, false);
+			}
```
Comment (mickaelistria) [inline org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacBindingResolver.java]: I don't think this change is related. Can you please move it away from this PR?

**Review by mickaelistria — CHANGES_REQUESTED** [review]
The JavacCompiler changes would fix https://github.com/eclipse-jdtls/eclipse-jdt-core-incubator/issues/839 . Can you please move the probably unrelated part about bindings?

File: org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacBindingResolver.java (hunk 1 above)
Comment (datho7561) [reply org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacBindingResolver.java]: working on it...

Comment (datho7561) [reply org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacBindingResolver.java]: see https://github.com/eclipse-jdtls/eclipse-jdt-core-incubator/pull/944

**Comment by mickaelistria** [comment]
Thanks!

**Pull request merged**